Update local_setup.bat.in

esteve opened the pull request:
Had to set it %~dp0 (which I don't know what it means, but @dirk-thomas showed me once), otherwise sourcing local_setup.bat results in the following error:

```
The system cannot find the path specified.
Environment variable AMENT_ENVIRONMENT_HOOK[ not defined
Environment variable AMENT_ENVIRONMENT_HOOK[ not defined
Environment variable AMENT_ENVIRONMENT_HOOK[ not defined
```

wjwwood commented:
+1

esteve commented:
I don't know what the change does, I'll leave it to @dirk-thomas to merge or modify this PR.

dirk-thomas commented:
`%~dp0` should expand to the absolute path of the directory where the current file is in. We need to test that when used in chained `.bat` files it expands to the correct path.
Also if the scripts always sets the value the following lines (7-11) are obsolete.

esteve commented:
Removed lines 8-11 and replaced them with ~%dp0, works fine locally. Triggered packaging job:

http://ci.ros2.org/view/packaging/job/ros2_packaging_windows/22/

dirk-thomas commented:
lgtm
